hub/api/tests/test_api.py

```diff
@@ -0,0 +1,75 @@
+import os, shutil
+import numpy as np
+from hub.api.dataset import Dataset
+from hub.core.chunk_engine.read import read_dataset_meta, read_tensor_meta
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def clear_test_files():
+    shutil.rmtree("/tmp/hub-test/", ignore_errors=True)
```

Comment:
in `pytest-options` (out of scope) we are calling "storage.clear()". could we add a method in API that does something like "ds.delete()"? i know this is a scary function but it's something i think we should still have. then after every test you can just call "ds.delete()".

for local provider, we overloaded the clear function to do exactly what youre doing here with shtuil.

Reply:
actually, maybe this is not necessary right now. i added another comment on another thread:

> also, could you add a TODO here for "create pytest Dataset fixture"? i would love for these Datasets to be created implicitly with the storage provider & all and you just write a test with "ds" as a parameter

and if we did this it would actually solve this problem

Reply:
because we can just do storage.clear() for the provider that Dataset is using & do it behind the scenes

Reply:
Problem is that part of the test coverage is the _constructor_. It's useful for some other tests maybe, but for some of these I really just want to make sure that it and the provider are playing nice and being set-up properly

Reply:
@McCrearyD done, as far as I could get it. Any additional expansion can be done in a separate PR. Please review.